### PR TITLE
Add Resource Properties API

### PR DESCRIPTION
**Pull request by jakecoll**

### CODE REVIEW

File: blazarclient/v1/hosts.py [hunk 1]
```diff
@@ -72,3 +72,40 @@ def reallocate(self, host_id, values):
...
+                                         key=lambda l: l[sort_by])
+        return resource_properties
+
+    def get_capability(self, capability_name):
+        resource_property = [
+            x for x in self.list_capabilities(detail=True)
+            if x['property'] == capability_name]
+
+        if resource_property:
+            del resource_property[0]['values']
```
Comment (diurnalist) [inline blazarclient/v1/hosts.py:101]: What if this `del` happens in `list_capabilities`? That seems a better place for it, after it is copied to its new key of 'capability_values'.

File: blazarclient/v1/hosts.py [hunk 2]
```diff
@@ -72,3 +72,40 @@ def reallocate(self, host_id, values):
...
+        resource_properties = body['resource_properties']
+
+        # Values is a reserved word in cliff so need to rename values column.
+        if detail:
+            for p in resource_properties:
+                p['capability_values'] = p['values']
+
+        if sort_by:
+            resource_properties = sorted(resource_properties,
+                                         key=lambda l: l[sort_by])
```
Comment (diurnalist) [inline blazarclient/v1/hosts.py:92]: I think `operator.itemgetter` does this for you:

```python
from operator import itemgetter

sorted(resource_properties, key=itemgetter(sort_by))
```

File: blazarclient/v1/shell_commands/hosts.py [hunk 3]
```diff
@@ -173,3 +174,59 @@ def args2body(self, parsed_args):
...
+        parser.add_argument(
+            '--sort-by', metavar="<extra_capability_column>",
+            help='column name used to sort result',
+            default='property'
+        )
+        return parser
+
+
+class UpdateHostCapability(command.UpdateCapabilityCommand):
+    """Update attributes of a host."""
```
Comment (diurnalist) [inline blazarclient/v1/shell_commands/hosts.py:228]:
```suggestion
    """Update attributes of a host capability."""
```

File: blazarclient/v1/shell_commands/networks.py [hunk 4]
```diff
@@ -172,3 +172,58 @@ def get_parser(self, prog_name):
...
+        parser.add_argument(
+            '--sort-by', metavar="<extra_capability_column>",
+            help='column name used to sort result',
+            default='property'
+        )
+        return parser
+
+
+class UpdateNetworkCapability(command.UpdateCapabilityCommand):
+    """Update attributes of a network."""
```
Comment (diurnalist) [inline blazarclient/v1/shell_commands/networks.py:226]:
```suggestion
    """Update attributes of a network capability."""
```

File: blazarclient/v1/hosts.py (hunk 2 above)
Comment (jakecoll) [reply blazarclient/v1/hosts.py:92]: the client uses lambda elsewhere so just being consistent for now.

**Pull request merged**